src/ray/gcs/gcs_server/gcs_node_manager.h

```diff
@@ -248,7 +248,7 @@ class GcsNodeManager : public rpc::NodeInfoHandler {
   /// Storage for GCS tables.
   std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage_;
   /// Raylet client pool.
-  std::shared_ptr<rpc::NodeManagerClientPool> raylet_client_pool_;
+  rpc::NodeManagerClientPool *raylet_client_pool_ = nullptr;
```

Comment:
This is never nullptr, so use`&`, not `*` ?

Reply:
afaik most style guides suggest pointers over references because of flexibility, moving, copying, it doesn't apply to this, but not bad to be consistent, but can we do absl::NotNull, I don't remember if we need cpp upgrade or absl upgrade for it?

Reply:
+1 for using reference if it's guaranteed to be not null.

Reply:
Because we have use cases passing nullptr: https://github.com/ray-project/ray/blob/ba8674ac946c39487714e49a2a26ed155143b7dd/src/ray/gcs/gcs_server/test/gcs_actor_scheduler_mock_test.cc#L41-L42